websites/R/Rumble/presence.ts

```diff
@@ -0,0 +1,193 @@
+const presence = new Presence({
+		clientId: "1090817608154108046",
+	}),
+	strings = presence.getStrings({
+		play: "presence.playback.playing",
+		pause: "presence.playback.paused",
+	}),
+
+ Assets = {
+	Rumble: "https://i.imgur.com/97X6RHQ.png",
+	Live: "https://i.imgur.com/LOh4SHJ.png",
+};
+presence.on("UpdateData", async () => {
+	// setup presence data
+	const presenceData: PresenceData = {
+		largeImageKey: Assets.Rumble,
+	};
+
+	// if page is a video
+	if (
+		document
+			.querySelectorAll("meta")[5]
+			.content.startsWith("https://rumble.com/v")
+	) {
+		// if video is live
+		if (document.querySelector(".chat--header")) {
+			(presenceData.smallImageKey = Assets.Live),
+				(presenceData.smallImageText = "Live");
+		}
+
+		// if in privacy mode
+		if (await presence.getSetting<string>("privacy")) {
+			// if video is live
+			if (presenceData.smallImageKey) 
+				presenceData.details = "Watching a livestream";
+			 else 
+				presenceData.details = "Watching a video";
+			
+
+			presence.setActivity(presenceData);
+			return;
+		}
+
+		// if show buttons is enabled
+		if (await presence.getSetting<string>("buttons")) {
+			presenceData.buttons = [
+				{
+					label: "Watch",
+					url: document.querySelectorAll("meta")[5].content,
+				},
+			];
+			// if video is live
+			if (presenceData.smallImageKey) 
+				presenceData.buttons[0].label += " Live";
+			 else 
+				presenceData.buttons[0].label += " Video";
+			
+		}
+
+		// if show channel picture is enabled
+		if (
+			document.querySelector("style").outerHTML.search("https://sp.rmbl.ws") &&
+			(await presence.getSetting<string>("channelPic"))
+		) {
+			// find pfp url in the stylesheet
+			presenceData.largeImageKey = document
+				.querySelector("style")
+				.outerHTML.substring(
+					document
+						.querySelector("style")
+						.outerHTML.search("https://sp.rmbl.ws/"),
+					document
+						.querySelector("style")
+						.outerHTML.substring(
+							document
+								.querySelector("style")
+								.outerHTML.search("https://sp.rmbl.ws/")
+						)
+						.search(";") +
+						document
+							.querySelector("style")
+							.outerHTML.search("https://sp.rmbl.ws/") -
+						1
+				);
+		}
+
+		presenceData.details =
+			document.querySelectorAll<HTMLVideoElement>(".h1")[0].innerText;
+		presenceData.state = document.querySelectorAll<HTMLVideoElement>(
+			".media-heading-name"
+		)[0].innerText;
```

Comment:
Note: this applies to multiple places
* `querySelectorAll(...)[0]` can be replaced with `querySelector(...)`
* Use `textContent` instead of `innerText`